x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/support/SecuritySystemIndices.java

```diff
@@ -124,13 +124,30 @@ private SystemIndexDescriptor getSecurityMainIndexDescriptor() {
             .setIndexPattern(".security-[0-9]+*")
             .setPrimaryIndex(MAIN_INDEX_CONCRETE_NAME)
             .setDescription("Contains Security configuration")
-            .setMappings(getMainIndexMappings())
+            .setMappings(getMainIndexMappings(INTERNAL_MAIN_INDEX_MAPPINGS_FORMAT))
             .setSettings(getMainIndexSettings())
             .setAliasName(SECURITY_MAIN_ALIAS)
             .setIndexFormat(INTERNAL_MAIN_INDEX_FORMAT)
             .setVersionMetaKey(SECURITY_VERSION_STRING)
             .setOrigin(SECURITY_ORIGIN)
             .setThreadPools(ExecutorNames.CRITICAL_SYSTEM_INDEX_THREAD_POOLS)
+            .setPriorSystemIndexDescriptors(
```

Comment:
Good job figuring this out! 

I was thinking initially that one of the nodes with a descriptor that's on the min version in the cluster should also be responsible for updating the mapping with the latest min version, that way we don't have to keep it around. Unfortunately the mapping update is only triggered when a prepare action happens (like in `prepareIndexIfNeededThenExecute`), so it depends on the descriptor mapping version on the node where the query is executed. 

If we want to switch over to use the `SystemIndexMappingUpdateService` for all security indices I think we have to take this approach since it always runs on the master node and we can't guarantee that the master node has the min mapping version.

Reply:
Can someone help to understand why we need to provide the prior index descriptors ? Shouldn't the logic always be install the new version as soon as possible (at least for passive changes such additions) ?

Reply:
The prior descriptors are needed so that if the index needs to be created, in a mixed-version cluster, the newer nodes know how to define the index in a way the older nodes can understand. The system index mappings version that is known by all nodes is negotiated as part of cluster formation (see `SystemIndexDescriptor.MappingsVersion`)